pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java

```diff
@@ -95,6 +95,7 @@ public void build(@Nullable SegmentVersion segmentVersion, ServerMetrics serverM
       reader = new RealtimeSegmentRecordReader(realtimeSegmentImpl, dataSchema, sortedColumn);
     }
     SegmentGeneratorConfig genConfig = new SegmentGeneratorConfig(dataSchema);
+    genConfig.setCheckTimeColumnValidityDuringGeneration(false);
```

Comment:
Good to add a comment here saying that we don't want realitme consumption to stop (if the exception fires, consumption _will_ stop) in case application is pushing bad data, so we explicitly ignore the time column check while generating realtime segments.

Reply:
Done.